boot-core/src/tests/unit_tests/mock.rs

```diff
@@ -0,0 +1,159 @@
+            cw20_base::msg::QueryMsg::Allowance { owner: _, spender: _ } => unimplemented!(),
+            cw20_base::msg::QueryMsg::AllAllowances { owner: _, start_after: _, limit: _ } => unimplemented!(),
+            cw20_base::msg::QueryMsg::AllSpenderAllowances { spender: _, start_after: _, limit: _ } => unimplemented!(),
+            cw20_base::msg::QueryMsg::AllAccounts { start_after: _, limit: _ } => unimplemented!(),
+            cw20_base::msg::QueryMsg::MarketingInfo {  } => unimplemented!(),
+            cw20_base::msg::QueryMsg::DownloadLogo {  } => unimplemented!(),
+        }
+    }
+
+    #[test]
+    fn mock() {
+        let sender = &Addr::unchecked(SENDER);
+        let recipient = &Addr::unchecked(BALANCE_ADDR);
+        let amount = 1000000u128;
+        let denom = "uosmo";
+
+        let mock = instantiate_default_mock_env(sender).unwrap();
+        let chain = mock.1;
+
+        chain.set_balance(recipient, vec![Coin::new(amount, denom)]).unwrap();
+        let balance = chain.query_balance(recipient, denom).unwrap();
+
+        asserting("address balance amount is correct")
+            .that(&amount).is_equal_to(&balance.into());
+
+        asserting("sender is correct")
+            .that(sender).is_equal_to(chain.sender());
+
+        let mut contract_source: ContractCodeReference = ContractCodeReference::default();
+
+        contract_source.contract_endpoints = Some(Box::new(
+            ContractWrapper::new(execute, instantiate, query)
+        ));
+
+        let init_res = chain.upload(&mut contract_source).unwrap();
+        asserting("contract initialized properly")
+            .that(&init_res.events[0].attributes[0].value)
+            .is_equal_to(&String::from("1"));
+
+        let init_msg = cw20_base::msg::InstantiateMsg {
+            name: String::from("Token"),
+            symbol: String::from("TOK"),
+            decimals: 6u8,
+            initial_balances: vec![],
+            mint: None,
+            marketing: None,
+        };
+        let init_res = chain.instantiate(1, &init_msg, None, Some(sender), &[]).unwrap();
+
+        let contract_address = Addr::unchecked(&init_res.events[0].attributes[0].value);
+
+        let exec_res = chain.execute(
+            &cw20_base::msg::ExecuteMsg::Mint {
+                recipient: recipient.to_string(),
+                amount: Uint128::from(100u128)
+            },
+            &[],
+            &contract_address)
+            .unwrap();
+
+        asserting("that exect passed on correctly")
+            .that(&exec_res.events[1].attributes[1].value)
+            .is_equal_to(&String::from("mint"));
+
+        let query_res = chain.query::<
+                cw20_base::msg::QueryMsg,
+                Response
+            >(&cw20_base::msg::QueryMsg::Balance {
+                address: recipient.to_string()
+            }, &contract_address).unwrap();
+
+        asserting("that query passed on correctly")
```

Comment:
if there is a `asserting!(..)` variant of this (notice the `!`) than prefer using that because it will add the line of the failed assertion to the test's output so you don't have to dig into your tests to find where it failed!

Reply:
ok!, changing that. first time using speculoos